main.py

```diff
@@ -79,6 +84,50 @@ def show_top_spending_category(df):
 
 
 
+def visualize_monthly_spending_trend(df):
+    print("--- Visualize Monthly Spending Trend ---")
+    # Get expense data, using copy to avoid affecting original df
+    expense_df = df[df["Type"] == "Expense"].copy()
+    # Convert Date column to datetime format
+    expense_df["Date"] = pd.to_datetime(expense_df["Date"])
+    # Filter month and add column called as Month
+    expense_df["Month"] = expense_df["Date"].dt.to_period("M")
+    # Group by Month to sum the spending
+    monthly_spending = expense_df.groupby("Month")["Amount"].sum().reset_index()
+    # For plotting, have to convert month back to datetime
+    monthly_spending["Month"] = monthly_spending["Month"].dt.to_timestamp()
```

Comment:
monthly_spending["Month"] includes day. For example, 2024-10-01. Please don't include day. So please change 2024-10-01 to 2024-10. You can check that it is good or not by adding another month data to csv.